### DISCUSSION TITLE
What's the expected behavior in wagmi when WalletConnect's `qrcode` option is set to false?

### DISCUSSION THREAD

**Discussion by nickbytes** [wevm/wagmi#152] [Q&A]
hi! 

I'm thinking through how one might customize the WalletConnect QR code in a dapp that uses wagmi.

If one was using the WalletConnect client, you might

```
import WalletConnect from '@walletconnect/client';

const connector = new WalletConnect({
  bridge: 'https://bridge.walletconnect.org',
});

if (connector && !connector.connected) {
  connector.createSession({ chainId: 1 }).then(() => {
    // do something with connector.uri, like create your own QR code
    console.log(connector.uri);
  });
}
```

Curious to know how one thinks about this within wagmi though?

Calling `connect` on wagmi's WalletConnect connector (demo below) when we've set `qrcode: false` currently does visibly does nothing (expected), but it's unclear to me how one could use wagmi's `useConnect` to access the connector's `uri` and allow users to copy/paste, created own QR code, etc etc. 

Demo link: https://codesandbox.io/s/nostalgic-liskov-ijlzl?file=/src/index.tsx
(this codesandbox is forked from previous @tmm link in Discussions, all I've changed is `qrcode: false`)

**Comment by tmm** [comment]
Good point! Setting `qrcode` to `false` ends up at a dead end.

You can pass a `connector` (like the one you created in your example) to WalletConnectConnector as an option, but wagmi isn't opinionated about what happens next.

Curious what you think the behavior should be. For example, maybe WalletConnectConnector's `connector.connect` only calls `provider.enable` if `qrcode === true`, maybe there's another option to customize the behavior, etc. I'm less familiar with using WalletConnect in "headless" mode so down to jam.

**Reply by nickbytes** [reply]
> For example, maybe WalletConnectConnector's connector.connect only calls provider.enable if qrcode === true, maybe there's another option to customize the behavior, etc.

Yea, this is interesting. So I'm sure you've seen that WalletConnect prints the `uri` to the console when the modal pops up. Wondering if in a "headless" mode, does it make sense for that `uri` to get returned from the `connect` call...? Maybe we're one of the few teams that has opted out of the WalletConnect modal experience in favor of our own, we've typically done this to create stylized QR codes.

We're working on creating an experience where all wallets are at the "top-level" rather than having to tap "WalletConnect" then seeing another modal/interface to scroll through, so we're effectively connecting to WalletConnect immediately and presenting a customized QR code interface.

**Reply by tmm** [reply]
Cool - so think I have a good grasp on how we can make this work.

Might be easiest to pair (or get set up with a basic example of generating a qr code). LMK which route sounds best :)

**Reply by nickbytes** [reply]
Definitely, will shoot you a message on twitter and start making a minimal repo example

**Reply by Chi111** [reply]
> Yea, this is interesting. So I'm sure you've seen that WalletConnect prints the `uri` to the console when the modal pops up. Wondering if in a "headless" mode, does it make sense for that `uri` to get returned from the `connect` call...? Maybe we're one of the few teams that has opted out of the WalletConnect modal experience in favor of our own, we've typically done this to create stylized QR codes.

hi, how to get uri? i am confused
